src/IO/S3Common.cpp

```diff
@@ -91,28 +95,294 @@ class AWSLogger final : public Aws::Utils::Logging::LogSystemInterface
     std::unordered_map<String, Poco::Logger *> tag_loggers;
 };
 
+class AWSEC2MetadataClient : public Aws::Internal::AWSHttpResourceClient
+{
+    static constexpr char EC2_SECURITY_CREDENTIALS_RESOURCE[] = "/latest/meta-data/iam/security-credentials";
```

Comment:
static constexpr const char *

Reply:
`error: ISO C++ forbids converting a string constant to 'char*' [-Werror=write-strings]`